app/(main)/recover-password/page.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { Button } from '@/components/Button/Button';
+import { Control, useForm, useWatch, SubmitHandler } from 'react-hook-form';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '../../../components/Form/Form';
+import { Input } from '@/components/Input/Input';
+import { useAuthContext } from '@/context/AuthContextProvider';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import LinkCustom from '@/components/LinkCustom/LinkCustom';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import Logo from '@/components/Logo/Logo';
+import logo from '@/public/assets/logo-colored-outline.svg';
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { recoverPassword } from '@/api/apiFunctions';
+import toast from 'react-hot-toast';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+
+/**
+ * The schema for the recover password form
+ * @throws {Error} Throws an error if the email is not a valid email address.
+ * @throws {Error} Throws an error if the email is not provided.
+ * @throws {Error} Throws an error if the email is not a string.
+ */
+const RecoverUserPasswordSchema = z.object({
+  email: z
+    .string()
+    .min(1, { message: 'Please enter an email address' })
+    .email({ message: 'Please enter a valid email address' }),
+});
+
+type ResetUserPasswordSchemaType = z.infer<typeof RecoverUserPasswordSchema>;
+
+/**
+ * Renders the recover password page.
+ * @returns {React.JSX.Element} The rendered recover password page.
+ */
+const RecoverPassword = (): React.JSX.Element => {
+  const router = useRouter();
+  const { isSignedIn, getUser } = useAuthContext();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (isSignedIn) {
+      getUser();
+      router.push('/league/all');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Don't need this. It's just a warning that's the IDE is complaining about.

Reply:
✅ fixed.